### PR TITLE
Downgrade duplicate nonce logs to debug

### LINKED ISSUE #4932: messagepool  print bug
2020-11-20T09:58:01.045+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100242", "newpremium": "125303", "addr": "f3q7cd5g7tqf7wnwatqr2hy76vsv2bmzzatfcnyszuv7brbpqn32yn6ky6h6375dyg3yijhs763ewm2aawphba", "nonce": 21402}
2020-11-20T09:58:01.068+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100321", "newpremium": "200642", "addr": "f3udt6gdw4jajc5ldwhdpkkrc3vlklt43gg5nwf6nxqaoqfce7puh4dmpnleaahiknqqv4n3k4ffnpbcm7ceta", "nonce": 20280}
2020-11-20T09:58:01.132+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100316", "newpremium": "125396", "addr": "f3thps4zeoaydau63z45ewgbgxurh2fhxo4kpq74dpkbuage42rb7qiom4b3j5wkh2adbxsbrnvhalf7vqkiba", "nonce": 118867}
2020-11-20T09:58:01.205+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "99899", "newpremium": "124874", "addr": "f3ugtwzstoxjwxizx2mmiwybmx5lhr4emz6tj2nvaoyfdkr7zboidjxmdmrctkaxeo3crdubfhw5ywuakh7f5a", "nonce": 99129}
2020-11-20T09:58:01.282+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "124453", "newpremium": "155567", "addr": "f3w2xf4h44st6qtwoncs2fzwjvpclzh3luaktjv4gchysjv74es62nh7n3m5spmpag3pawjfprieop7yfyiraa", "nonce": 96147}
2020-11-20T09:58:01.519+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100899", "newpremium": "126124", "addr": "f3rmhlmqfaph6czwiqwlg3kfjgejugt5thcviowlmt3l42464q25ptk3znphuuiwrdbyumun3ui7q2gut7v2da", "nonce": 239155}
2020-11-20T09:58:01.650+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "246406", "newpremium": "308008", "addr": "f3uoxqkxhzirhxez7lfvhkkkhmeed4nq4kkqthn7pguam4rtbb2nop6fnzbdmrpetdhwlgzznbid2rbobtsw3a", "nonce": 57220}
2020-11-20T09:58:02.019+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100197", "newpremium": "125247", "addr": "f3rq5kpynijbb5mmuo6om72d556qjpofhlk4n5ajoawrrwqd7w5vg5mc7iqwlhgxjbfcopoktvl2h6g3kjb5na", "nonce": 11286}
2020-11-20T09:58:02.113+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100591", "newpremium": "201182", "addr": "f3xbhu7ukbh2uwy6ayzgtb2xwevdn6mpiiyfnpczjmt6rvpnemyu542dvboqw24i2nfcnw5vxvh2hp76v77ogq", "nonce": 245175}
2020-11-20T09:58:02.237+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100306", "newpremium": "125383", "addr": "f3vccjwrbqfu64arsar2frk6beqk6hp76jn7zoc4eahayh7r4dt6hrdpybanwgmzqzjmrprceow5tzcmygyglq", "nonce": 43921}
2020-11-20T09:58:02.333+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100332", "newpremium": "125416", "addr": "f3thps4zeoaydau63z45ewgbgxurh2fhxo4kpq74dpkbuage42rb7qiom4b3j5wkh2adbxsbrnvhalf7vqkiba", "nonce": 118868}
2020-11-20T09:58:02.565+0800	INFO	messagepool	messagepool/messagepool.go:246	add with duplicate nonce
2020-11-20T09:58:02.591+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100666", "newpremium": "536633055", "addr": "f3rvsqfarxcs7o5us2nd2tgaq3dwbjeodjwit5zsj4uy2mlk632a5ud2rvzujxrg6byid6yaw2iogaxaxg4q5q", "nonce": 24412}
2020-11-20T09:58:02.653+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100745", "newpremium": "125932", "addr": "f3rmhlmqfaph6czwiqwlg3kfjgejugt5thcviowlmt3l42464q25ptk3znphuuiwrdbyumun3ui7q2gut7v2da", "nonce": 239156}
2020-11-20T09:58:02.678+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100562", "newpremium": "125703", "addr": "f3w4b3qu6vnmdlgjbk76vul33nqcblzzaq2bc6drz7f2zkndfashxhdjj6tthec2khckfhtdrzjiiuvymbo2vq", "nonce": 10678}
2020-11-20T09:58:02.769+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100277", "newpremium": "125347", "addr": "f3ugtwzstoxjwxizx2mmiwybmx5lhr4emz6tj2nvaoyfdkr7zboidjxmdmrctkaxeo3crdubfhw5ywuakh7f5a", "nonce": 99130}
2020-11-20T09:58:02.951+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "125658", "newpremium": "157073", "addr": "f3uzd3uq4iydpdh4btejvxqhe4giwszccezqdsitkovgyxxvil7chxzwlr233rqfd5vspnoptueqbhf2eb7via", "nonce": 9100}
2020-11-20T09:58:03.125+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "99978", "newpremium": "124973", "addr": "f3r5kqidwlhqfziqjoiziymkkexfybmi26h5ezvz4zejratpetej66bfgxiv4hljtbeven7itimhswfc45gyha", "nonce": 79423}
2020-11-20T09:58:03.162+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100141", "newpremium": "200282", "addr": "f3xbhu7ukbh2uwy6ayzgtb2xwevdn6mpiiyfnpczjmt6rvpnemyu542dvboqw24i2nfcnw5vxvh2hp76v77ogq", "nonce": 245176}
2020-11-20T09:58:03.330+0800	INFO	messagepool	messagepool/messagepool.go:246	**add with duplicate nonce**
2020-11-20T09:58:03.333+0800	INFO	messagepool	messagepool/messagepool.go:246	**add with duplicate nonce**
2020-11-20T09:58:03.348+0800	INFO	messagepool	messagepool/messagepool.go:246	**add with duplicate nonce**
2020-11-20T09:58:03.376+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100331", "newpremium": "125414", "addr": "f3q7cd5g7tqf7wnwatqr2hy76vsv2bmzzatfcnyszuv7brbpqn32yn6ky6h6375dyg3yijhs763ewm2aawphba", "nonce": 21403}
2020-11-20T09:58:03.456+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100330", "newpremium": "125413", "addr": "f3rq5kpynijbb5mmuo6om72d556qjpofhlk4n5ajoawrrwqd7w5vg5mc7iqwlhgxjbfcopoktvl2h6g3kjb5na", "nonce": 11287}
2020-11-20T09:58:03.511+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100876", "newpremium": "126096", "addr": "f3thps4zeoaydau63z45ewgbgxurh2fhxo4kpq74dpkbuage42rb7qiom4b3j5wkh2adbxsbrnvhalf7vqkiba", "nonce": 118869}
2020-11-20T09:58:03.588+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100465", "newpremium": "125582", "addr": "f3vj53sqfqnqpgbqfmxav3hqac43nyojwetnnvuelaib6y2l2ygm3xa7h2qgkrt2tnrgtgjye5cmpnjj3uwd3q", "nonce": 13511}
2020-11-20T09:58:03.754+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "100283", "newpremium": "125354", "addr": "f3rmhlmqfaph6czwiqwlg3kfjgejugt5thcviowlmt3l42464q25ptk3znphuuiwrdbyumun3ui7q2gut7v2da", "nonce": 239157}
2020-11-20T09:58:03.838+0800	INFO	messagepool	messagepool/messagepool.go:243	add with RBF	{"oldpremium": "99935", "newpremium": "124919", "addr": "f3vccjwrbqfu64arsar2frk6beqk6hp76jn7zoc4eahayh7r4dt6hrdpybanwgmzqzjmrprceow5tzcmygyglq", "nonce": 43922}

### PR DESCRIPTION
**Pull request by zgfzgf**
fix #4932

### CODE REVIEW

**Review by vyzo — CHANGES_REQUESTED** [review]
Please don't change the grouping of imports.

Also, what exactly does this fix?

**Comment by zgfzgf** [comment]
Should have been printed, not printed
and Printing too much useless information

File: chain/messagepool/messagepool.go [hunk 1]
```diff
@@ -240,10 +240,13 @@ func (ms *msgSet) add(m *types.SignedMessage, mp *MessagePool, strict, untrusted
...
 			if types.BigCmp(m.Message.GasPremium, minPrice) >= 0 {
-				log.Infow("add with RBF", "oldpremium", exms.Message.GasPremium,
+				log.Debugw("add with RBF", "oldpremium", exms.Message.GasPremium,
 					"newpremium", m.Message.GasPremium, "addr", m.Message.From, "nonce", m.Message.Nonce)
 			} else {
-				log.Info("add with duplicate nonce")
+				log.Infof("add with duplicate nonce. message from %s with nonce %d already in mpool,"+
+					" increase GasPremium to %s from %s to trigger replace by fee: %w",
+					m.Message.From, m.Message.Nonce, minPrice, m.Message.GasPremium,
+					ErrRBFTooLowPremium)
```
Comment (vyzo) [inline chain/messagepool/messagepool.go:249]: there is no reason to construct a wrapped error for this log, just log what you need; Also note you are repeating the information from the error, which will get logged together with the previous log message.

So I don't see this as adding anything.

File: chain/messagepool/messagepool.go [hunk 2]
```diff
@@ -240,10 +240,13 @@ func (ms *msgSet) add(m *types.SignedMessage, mp *MessagePool, strict, untrusted
 			// check if RBF passes
 			minPrice := ComputeMinRBF(exms.Message.GasPremium)
 			if types.BigCmp(m.Message.GasPremium, minPrice) >= 0 {
-				log.Infow("add with RBF", "oldpremium", exms.Message.GasPremium,
+				log.Debugw("add with RBF", "oldpremium", exms.Message.GasPremium,
```
[the remark this answers is not in the document]
Comment (magik6k) [reply chain/messagepool/messagepool.go:243]: TBH both of those log messages should be Debug, they aren't that useful to the user (not actionable in any way), and are really spammy (90% of my logs are coming from here)

[the remark this answers is not in the document]
Comment (vyzo) [reply chain/messagepool/messagepool.go:243]: ok, let's then turn both of them to DEBUGs and make it useful.

[the remark this answers is not in the document]
Comment (zgfzgf) [reply chain/messagepool/messagepool.go:243]: now from info to debug

File: chain/messagepool/messagepool.go [hunk 3]
```diff
@@ -240,10 +240,13 @@ func (ms *msgSet) add(m *types.SignedMessage, mp *MessagePool, strict, untrusted
...
 					"newpremium", m.Message.GasPremium, "addr", m.Message.From, "nonce", m.Message.Nonce)
 			} else {
-				log.Info("add with duplicate nonce")
+				log.Debugf("add with duplicate nonce. message from %s with nonce %d already in mpool,"+
+					" increase GasPremium to %s from %s to trigger replace by fee: %w",
```
Comment (vyzo) [inline chain/messagepool/messagepool.go:247]: this `%w` formatting directive is invalid in the log, it's only valid in the error wrapping function; it should be `%s`.

Comment (zgfzgf) [reply chain/messagepool/messagepool.go]: hi @vyzo thank you

Comment (Kubuxu) [reply chain/messagepool/messagepool.go]: FYI to all, the best format to use for errors is `%v`, or if you want detailed info `%+v`.

Comment (vyzo) [reply chain/messagepool/messagepool.go]: it's not necessary here - it's not a structured error.

**Comment by vyzo** [comment]
Can we squash this down to one commit? It's really messy for something that changes two incosequential lines of code.

**Review by magik6k — APPROVED** [review]

**Pull request merged**